keadm/cmd/keadm/app/cmd/debug/get.go

```diff
@@ -0,0 +1,1017 @@
+			return err
+		}
+		if _, err := fmt.Fprintln(out); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// PrintGeneric Output object data to out stream through printer
+func PrintGeneric(printer printers.ResourcePrinter, obj runtime.Object, out io.Writer) error {
+	isList := meta.IsListType(obj)
+	if isList {
+		items, err := meta.ExtractList(obj)
+		if err != nil {
+			return err
+		}
+
+		// take the items and create a new list for display
+		list := &unstructured.UnstructuredList{
+			Object: map[string]interface{}{
+				"kind":       "List",
+				"apiVersion": "v1",
+				"metadata":   map[string]interface{}{},
+			},
+		}
+		if listMeta, err := meta.ListAccessor(obj); err == nil {
+			list.Object["metadata"] = map[string]interface{}{
+				"selfLink":        listMeta.GetSelfLink(),
+				"resourceVersion": listMeta.GetResourceVersion(),
+			}
+		}
+
+		for _, item := range items {
+			list.Items = append(list.Items, *item.(*unstructured.Unstructured))
+		}
+		if err := printer.PrintObj(list, out); err != nil {
+			return err
+		}
+	} else {
+		var value map[string]interface{}
+		data, err := json.Marshal(obj)
+		if err != nil {
+			return err
+		}
+		if err := json.Unmarshal(data, &value); err != nil {
+			return err
+		}
+		if err := printer.PrintObj(&unstructured.Unstructured{Object: value}, out); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ConvertDataToTable Convert the data into table kind to simulate the data sent by api-server
+func ConvertDataToTable(obj runtime.Object) (runtime.Object, error) {
+	to := metav1.TableOptions{}
+	tc := storage.TableConvertor{TableGenerator: k8sprinters.NewTableGenerator().With(printersinternal.AddHandlers)}
+
+	return tc.ConvertToTable(context.TODO(), obj, &to)
+}
+
+// ParseMetaToAPIList Convert the data to the corresponding list type according to the apiserver usage type
+// Only use this type definition to get the table header processing handle,
+// and automatically obtain the ColumnDefinitions of the table according to the type
+// Only used by HumanReadablePrint.
+func ParseMetaToAPIList(results []dao.Meta) (*api.PodList, *api.ServiceList, *api.SecretList, *api.ConfigMapList, *api.EndpointsList, error) {
+	podList := &api.PodList{}
+	serviceList := &api.ServiceList{}
+	secretList := &api.SecretList{}
+	configMapList := &api.ConfigMapList{}
+	endPointsList := &api.EndpointsList{}
+	value := make(map[string]interface{})
+
+	for _, v := range results {
+		switch v.Type {
+		case ResourceTypePod:
+			pod := api.Pod{}
+
+			if err := json.Unmarshal([]byte(v.Value), &value); err != nil {
+				return nil, nil, nil, nil, nil, err
+			}
+			metadata, err := json.Marshal(value["metadata"])
+			if err != nil {
+				return nil, nil, nil, nil, nil, err
+			}
+			spec, err := json.Marshal(value["spec"])
+			if err != nil {
+				return nil, nil, nil, nil, nil, err
+			}
+			status, err := json.Marshal(value["status"])
+			if err != nil {
+				return nil, nil, nil, nil, nil, err
+			}
```

Comment:
This code is repeated a lot in each cace, try to extract it into a method to reuse